### ISSUE TITLE
Duplicate collection generated when creating new API version

### ISSUE THREAD

**Issue by br-tyler-milner** [a85/POSTMan-Chrome-Extension#10289]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the Issue

When creating a new API version, adding existing API elements like a documentation collection causes duplicates of those collections to be generated.
![image](https://user-images.githubusercontent.com/78233128/132769774-5eb3a0fb-5b69-4634-9695-a151bc278057.png)

The duplicate collection that gets generated has the new API version appended to the end of the collection name:
![api-duplicate-collection](https://user-images.githubusercontent.com/78233128/132770365-ae8ebd85-b60f-436e-b172-db4048603b27.png)

This will lead to significant collection clutter in the workspace, especially as API iterations happen involving other collection types like mocking, testing, etc.
![collection-duplicate-collection](https://user-images.githubusercontent.com/78233128/132770405-dd39507b-46db-4261-a814-4be100f2c2be.png)

If you don't add existing API elements when creating the new API version, then no duplicate collection is generated as expected, but you lose the ability to add the existing documentation collection to the new API version. A "no collections in this workspace" message is shown, despite the original collection continuing to exist:
![api-no-collections-in-workspace](https://user-images.githubusercontent.com/78233128/132772015-ab068a38-f30d-4144-b748-cc1a8dbd3b49.png)

The way this worked previously, the existing collection would be associated with the new API version (i.e. no duplicate collection would be generated). From there, you could apply changes to the existing collection and then "tag" your changes to correspond to the new API version.

This resulted in a single collection being iterated on and easily kept up to date with new API versions using the "validation" functionality built in to Postman. That workflow is now broken since a brand new collection is generated for each new API version. This means that you lose the ability to keep a single collection containing a changelog that corresponds to the updates made to the API over time. Aside from preventing collection bloat, one of the main benefits to having a single collection was that the documentation URL didn't change. Now the duplicate collection that's generated has a new documentation URL and the user loses the ability to browse the various tags/versions of the API from a single documentation webpage.

### Steps To Reproduce

1. Create an API.
2. Generate a collection from that API (e.g. a "documentation" collection).
3. (Optional) Tag the collection to the appropriate version of the API (to keep everything versioned and in sync).
4. Create a new API version, selecting the collection in the list of "elements to add".
5. Observe a duplicate of the selected collection has been created, with the new API version appended to the end of its name.

### Screenshots or Videos

https://user-images.githubusercontent.com/78233128/132772671-e67ebdba-d99a-4c04-b555-25fa0db27b16.mov

### Environment Information

```markdown
- Operating System: macOS 11.5.2
- Platform Type: Native App
- Postman Version: 8.12.1
```

### Additional Context?

Unfortunately, this isn't the only thing that's broken involving API/collection versioning recently. I've been following #10271, which prevented collections from being tagged at all. I'm guessing that this duplication bug could be related to the mentioned changes that leaked out.

**Comment by rochanram** [comment]
This will be the behaviour going forward. Whenever a new version is created, you can either choose not to carry forward any of the relations like the documentation or tests. And in case you choose to carry forward any of the relations from the previous version, it will be duplicated.

**Comment by br-tyler-milner** [comment]
@rochanram can you comment on why this change in behavior was decided? It's now going to create significant problems with regard to collection bloat as an API is designed/built from the ground-up (i.e. API-first approach). It's not hard to imagine this leading to 10's or even 100's of collections being added to a workspace over time, which will undoubtably cause slowdowns, confusion, and frustration.

Also, when I create a new API version, how can I easily apply those changes to the documentation collection? With this new approach, it seems to be enforcing a manual process as opposed to using the API-collection validation functionality that's built in to Postman. Again, this will now lead to an error prone, slow, and frustrating process of keeping API documentation up to date as an API specification evolves over time.

**Comment by abhijitkane** [comment]
@br-tyler-milner The latest release (Postman v9) changes this flow a little bit. Versions now represent parallel lines of development for an API, and releases represent the state of the API version at a point in time. You can also "view" releases (this will show you snapshots of the API definition and any collections in the release).

For the workflow you discussed above, you'll need to create a release on your API version (described here - https://learning.postman.com/docs/designing-and-developing-your-api/versioning-an-api/#working-with-api-releases). Once this is done, you'll be able to create documentation/mock servers/monitors by specifying any of the release tags created on the collection. Version tags that have already been created in older versions of Postman will continue to show up while creating documentation.

We'd love to hear your thoughts on the new workflow once you've had a chance to try it out. A more complete explanation of changes can be found at https://blog.postman.com/the-new-postman-api-platform/.

**Comment by br-tyler-milner** [comment]
@abhijitkane thanks for the information. After reading the documentation and doing a bit of experimenting, I've come up with the following workflow for both initial creation of the API as well as iteration on the contract. Could you please validate?

This assumes you're starting fresh with a brand new app/API that needs to be defined, developed, tested, and iterated on in an API-First manner.

**Creating the API:**
1. Create an API in Postman and give it a version (e.g. "v0").
2. Create/edit the OpenAPI schema/YAML as necessary.
3. Create documentation for the API. This automatically generates an associated documentation collection in the workspace.
4. Create a "release" for the API (e.g. "0.1.0").

**Updating the API:**
1. Edit the API's OpenAPI schema/YAML as necessary.
2. Make updates to the associated documentation collection, using the "validation" functionality or applying changes manually as necessary.
3. Create a "release" for the API (e.g. "0.1.1").

To me, it seems like the main differences in this updated workflow are:
* I'm no longer manually "tagging" updates to the documentation collection. The documentation collection appears to be automatically tagged when I create a new API "release".
* While in the initial design/development phase, even though things are changing very rapidly and breaking changes are occasionally being introduced, I'm always creating an API "release" instead of a new API "version". By following some naming conventions for the release names, I can help to signify breaking vs non-breaking changes (e.g. "0.1.0" vs "0.1.1"). It sounds like the only time I'd create a new API "version" is when I want to completely overhaul the API with a totally new endpoint/resource structure after already shipping an API to production (e.g. "v2" vs "v1").

Does all that sound correct from your perspective?

**Comment by abhijitkane** [comment]
@br-tyler-milner Yep, the workflow you described makes sense. The tagging is now automatic - all collections and the schema within the API version are tagged when the release is created. Releases can be created as often as necessary, of course (and the 0.1.0 vs 0.1.1 convention for breaking-vs-nonbreaking makes sense).

Happy to chat more about this to understand your workflow better. We want to make sure this updated product evolves to support the way you develop and maintain your internal/external APIs. Feel free to schedule a call via https://calendly.com/abhijit_kane/30min anytime.

**Comment by br-tyler-milner** [comment]
The new workflow described above using API "releases" rather than explicit API "versions" is working for me now. Closing issue...
